### PR TITLE
Queue Submit prework...

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
Some cleanups in the syncval code base to enable queue submit validation (in a separate PR) and it to improve the signal-to-noise for review of the later PR.

### CODE REVIEW

File: layers/synchronization_validation.cpp [hunk 1]
```diff
@@ -202,16 +202,15 @@ static std::string string_SyncStageAccessFlags(const SyncStageAccessFlags &flags
...
 }
 
-static std::string string_UsageTag(const ResourceUsageRecord &tag) {
-    std::stringstream out;
-
-    out << "command: " << CommandTypeString(tag.command);
-    out << ", seq_no: " << tag.seq_num;
-    if (tag.sub_command != 0) {
-        out << ", subcmd: " << tag.sub_command;
+std::ostream &operator<<(std::ostream &out, const ResourceUsageRecord &record) {
```
Comment (jeremyg-lunarg) [inline layers/synchronization_validation.cpp:205]: making LogError() etc be stream-like would be really nice, so yay for stepping in that direction.

**Review by jeremyg-lunarg — COMMENTED** [review]
Most of this definitely LGTM. But I haven't fully understood the replay & tagging commits:

syncval: Revise subpass tagging to isolate syncops 
syncval: Cleanup of invalid tagging 
syncval: Cleanup/infrastruct. for RP syncop replay

They're quite a bit more complex than the others. I'll look at them again later this afternoon.

File: layers/synchronization_validation.cpp [hunk 2]
```diff
@@ -220,6 +219,50 @@ static std::string string_UsageIndex(SyncStageAccessIndex usage_index) {
     return std::string(stage_access_name);
 }
 
+struct SyncNodeFormatter {
```
Comment (jzulauf-lunarg) [inline layers/synchronization_validation.cpp:222]: @jeremyg-lunarg -- what do you think of think pattern.  In this case, what I need to stream has several parameters, so I define a container with a dedicate stream out overload.

From a style point of view one could implement the formatting as part of the container -- then define only a single << override a FormatterBase (with a virtual "do the formatting" function)

Probably want to explore before scaling.

```

Comment (jeremyg-lunarg) [reply layers/synchronization_validation.cpp:222]: I assume there's N more uses coming in future PRs. As long as that's the case it seems like win to me.

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM. A docs update to explain how replay works would be really helpful, once all the phase2 stuff is nailed down enough to be documentable...

**Pull request merged**